From b5ce701eefb177b2e6fc59229cd9923391413970 Mon Sep 17 00:00:00 2001
From: Mohammed Sameer <msameer@foolab.org>
Date: Mon, 14 Apr 2014 22:26:45 +0300
Subject: [PATCH 5/6] Keep video branch in NULL state

This patch has been extracted rom Sailfish OS public source code release
and ported to GStreamer 1.x
Original patch: 0004-Keep-video-branch-in-NULL-state.patch
---
 gst/camerabin2/gstcamerabin2.c | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/gst/camerabin2/gstcamerabin2.c b/gst/camerabin2/gstcamerabin2.c
index 8638859..5a9ee90 100644
--- a/gst/camerabin2/gstcamerabin2.c
+++ b/gst/camerabin2/gstcamerabin2.c
@@ -410,6 +410,15 @@ gst_camera_bin_start_capture (GstCameraBin2 * camerabin)
     g_free (location);
   }
 
+  if (camerabin->mode == MODE_VIDEO) {
+    if (camerabin->video_filter) {
+      gst_element_sync_state_with_parent (camerabin->video_filter);
+    }
+
+    gst_element_sync_state_with_parent (camerabin->videobin_capsfilter);
+    gst_element_sync_state_with_parent (camerabin->video_encodebin);
+  }
+
   g_signal_emit_by_name (camerabin->src, "start-capture", NULL);
   if (camerabin->mode == MODE_VIDEO) {
     camerabin->audio_send_newseg = TRUE;
@@ -980,10 +989,15 @@ gst_camera_bin_video_reset_elements (gpointer u_data)
   gst_element_set_state (camerabin->videobin_capsfilter, GST_STATE_NULL);
   if (camerabin->video_filter) {
     gst_element_set_state (camerabin->video_filter, GST_STATE_NULL);
+#if 0
     gst_element_sync_state_with_parent (camerabin->video_filter);
+#endif
   }
+
+#if 0
   gst_element_sync_state_with_parent (camerabin->videobin_capsfilter);
   gst_element_sync_state_with_parent (camerabin->video_encodebin);
+#endif
 
   if (camerabin->audio_src) {
     gst_element_set_state (camerabin->audio_capsfilter, GST_STATE_READY);
@@ -1653,6 +1667,7 @@ gst_camera_bin_create_elements (GstCameraBin2 * camera)
      */
     gst_element_set_locked_state (camera->videosink, TRUE);
     gst_element_set_locked_state (camera->imagesink, TRUE);
+    gst_element_set_locked_state (camera->video_encodebin, TRUE);
 
     g_object_set (camera->videosink, "location", camera->location, NULL);
     g_object_set (camera->imagesink, "location", camera->location, NULL);
-- 
1.8.5.2

